zipline/data/hdf5_daily_bars.py

```diff
@@ -470,6 +479,27 @@ def _compute_date_range_slice(self, start_date, end_date):
 
         return slice(start_ix, end_ix)
 
+    def _validate_assets(self, assets):
+        """Validate that asset identifiers are contained in the daily bars.
+
+        Parameters
+        ----------
+        assets : array-like[int]
+           The asset identifiers to validate.
+
+        Raises
+        ------
+        NoDataForSid
+            If one or more of the provided asset identifiers are not
+            contained in the daily bars.
+        """
+        if not np.in1d(assets, self.sids).all():
+            raise NoDataForSid()
```

Comment:
we could use `np.setdiff1d` instead of `in1d` and then report the sids that are missing. That might be useful for the vectorized case if only one or two assets are missing in a large array.

Reply:
Good idea, will update